modules/core/src/smithy4s/Validator.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright 2021-2024 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+
+trait Validator[A, B] {
```

Comment:
**question:** can/should we make this sealed, to allow ourselves some API changes? AFAIU users aren't expected to instantiate these, apart from `of` and the combinators like `validating`.

Reply:
Done.